autotst/job/job.py

```diff
@@ -58,13 +60,29 @@ def __init__(
 
         self.calculator = calculator
         if self.calculator:
-            self.directory = self.calculator.directory
+            if directory is None:
+                logging.info("Job directory not specified...setting Job directory to calculator directory")
+                self.directory = self.calculator.directory
+            else:
+                logging.info("Setting calculator directory to Job directory")
+                self.directory = self.calculator.directory = directory
+            if scratch is None:
+                logging.info("Job scratch directory not specified...setting Job scratch to calculator scratch")
+                self.scratch = self.calculator.scratch
+            else:
+                logging.info("Setting calculator scratch to Job scratch")
+                self.scratch = self.calculator.scratch = scratch
+        
```

Comment:
What if there is no calculator and no directory / no scratch. We should probably have an "else" and set both to "." if not specified